dotnetOQS/KEM.cs

```diff
@@ -74,15 +98,33 @@ static KEM()
             // initialize list of supported/enabled mechanisms
             List<string> enabled = new List<string>();
             List<string> supported = new List<string>();
-            int count = OQS_KEM_alg_count();
-            for (int i = 0; i < count; i++)
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))

```

Comment:
It would be nice to avoid these branching calls all over the program. Would that go away if the DLL was named something different?

Reply:
Yeah, the file name limitations currently. Until there is a better way to resolve this, I am unable to point a different file name during a compile time constant for DllImport

Reply:
I think I found a solution for this problem. I have to bump the minimum to 1.5 to use a better solution with little impact on compatibility.

Reply:
I have found out that it actually prepend "lib" on library names on Linux and macOS. Oh well, I have undo them and just edit the DllImport constant to "oqs".